### PR TITLE
fix(gpu): fix compression

### PR DESCRIPTION
**Pull request by pdroalves**
Fixes some edge cases, including booleans and multi-gpu.

closes: https://github.com/zama-ai/tfhe-rs-internal/issues/686

### Check-list:

* [ ] Tests for the changes have been added (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Relevant issues are marked as resolved/closed, related issues are linked in the description
* [ ] Check for breaking changes (including serialization changes) and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

**Comment by pdroalves** [comment]
Now we can run compression benchmarks using the target `bench_integer_compression_gpu` and tests using `test_integer_compression_gpu`.

File: backends/tfhe-cuda-backend/cuda/src/integer/compression/compression.cu [hunk 1]
```diff
@@ -25,24 +25,25 @@ void scratch_cuda_integer_decompress_radix_ciphertext_64(
     uint32_t compression_glwe_dimension, uint32_t compression_polynomial_size,
     uint32_t lwe_dimension, uint32_t pbs_level, uint32_t pbs_base_log,
     uint32_t num_lwes, uint32_t message_modulus, uint32_t carry_modulus,
-    PBS_TYPE pbs_type, uint32_t storage_log_modulus, bool allocate_gpu_memory) {
+    PBS_TYPE pbs_type, uint32_t storage_log_modulus, uint32_t bodies_count,
```
Comment (agnesLeroy) [inline backends/tfhe-cuda-backend/cuda/src/integer/compression/compression.cu:28]: Is bodies_count used? I doesn't seem so to me? :thinking:

**Review by agnesLeroy — COMMENTED** [review]
Hey @pdroalves! Thanks a lot for this fix: I just have one question, just to check the additional argument bodies_count is necessary :slightly_smiling_face:

File: backends/tfhe-cuda-backend/cuda/src/integer/compression/compression.cu (hunk 1 above)
Comment (pdroalves) [reply backends/tfhe-cuda-backend/cuda/src/integer/compression/compression.cu:28]: Yes, we need it to unpack the input GLWE in decompression. I was initially trying to compute it inside the function but I concluded it was easier to just pass it from rust side.

However there is a inconsistency in naming. Some places we call it "bodies" and in other "body". I will refactor.

**Review by agnesLeroy — APPROVED** [review]
We still have body on the c++ side vs bodies on the Rust side, but that's not really so much of an issue I'll merge it as it is I think.
